src/modules/hip/kernel/box_filter.hpp

```diff
@@ -0,0 +1,775 @@
+    dst_f8->x.w = fmaf(src_f, 0.01234568f, dst_f8->x.w);
+    dst_f8->y.x = fmaf(src_f, 0.01234568f, dst_f8->y.x);
+    dst_f8->y.y = fmaf(src_f, 0.01234568f, dst_f8->y.y);
+    dst_f8->y.z = fmaf(src_f, 0.01234568f, dst_f8->y.z);
+    dst_f8->y.w = fmaf(src_f, 0.01234568f, dst_f8->y.w);
+    src_f = rpp_hip_unpack3(src);
+    dst_f8->x.w = fmaf(src_f, 0.01234568f, dst_f8->x.w);
+    dst_f8->y.x = fmaf(src_f, 0.01234568f, dst_f8->y.x);
+    dst_f8->y.y = fmaf(src_f, 0.01234568f, dst_f8->y.y);
+    dst_f8->y.z = fmaf(src_f, 0.01234568f, dst_f8->y.z);
+    dst_f8->y.w = fmaf(src_f, 0.01234568f, dst_f8->y.w);
+    src = src_uchar4[3];
+    src_f = rpp_hip_unpack0(src);
+    dst_f8->y.x = fmaf(src_f, 0.01234568f, dst_f8->y.x);
+    dst_f8->y.y = fmaf(src_f, 0.01234568f, dst_f8->y.y);
+    dst_f8->y.z = fmaf(src_f, 0.01234568f, dst_f8->y.z);
+    dst_f8->y.w = fmaf(src_f, 0.01234568f, dst_f8->y.w);
+    src_f = rpp_hip_unpack1(src);
+    dst_f8->y.y = fmaf(src_f, 0.01234568f, dst_f8->y.y);
+    dst_f8->y.z = fmaf(src_f, 0.01234568f, dst_f8->y.z);
+    dst_f8->y.w = fmaf(src_f, 0.01234568f, dst_f8->y.w);
+    src_f = rpp_hip_unpack2(src);
+    dst_f8->y.z = fmaf(src_f, 0.01234568f, dst_f8->y.z);
+    dst_f8->y.w = fmaf(src_f, 0.01234568f, dst_f8->y.w);
+    src_f = rpp_hip_unpack3(src);
+    dst_f8->y.w = fmaf(src_f, 0.01234568f, dst_f8->y.w);
+}
+
+// Handles PKD3->PKD3 for any combination of kernelSize = 3/5/7/9 and T = U8/F32/F16/I8
+template <typename T>
+__global__ void box_filter_pkd_tensor(T *srcPtr,
+                                      uint2 srcStridesNH,
+                                      T *dstPtr,
+                                      uint2 dstStridesNH,
+                                      uint kernelSize,
+                                      uint padLength,
+                                      uint2 tileSize,
+                                      RpptROIPtr roiTensorPtrSrc)
+{
+    int hipThreadIdx_x8 = hipThreadIdx_x << 3;
+    int id_x_o = (hipBlockIdx_x * tileSize.x * 8) + hipThreadIdx_x8;
+    int id_y_o = hipBlockIdx_y * tileSize.y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    int id_x_i = id_x_o - padLength;
+    int id_y_i = id_y_o - padLength;
+    d_float24 sum_f24;
+    __shared__ uchar src_lds[48][128];
+
+    int srcIdx = (id_z * srcStridesNH.x) + ((id_y_i + roiTensorPtrSrc[id_z].xywhROI.xy.y) * srcStridesNH.y) + ((id_x_i + roiTensorPtrSrc[id_z].xywhROI.xy.x) * 3);
+    int dstIdx = (id_z * dstStridesNH.x) + (id_y_o * dstStridesNH.y) + id_x_o * 3;
+    sum_f24.x.x = (float4) 0;
+    sum_f24.x.y = (float4) 0;
+    sum_f24.y.x = (float4) 0;
+    sum_f24.y.y = (float4) 0;
+    sum_f24.z.x = (float4) 0;
+    sum_f24.z.y = (float4) 0;
+
+    int3 hipThreadIdx_y_channel;
+    hipThreadIdx_y_channel.x = hipThreadIdx_y;
+    hipThreadIdx_y_channel.y = hipThreadIdx_y + 16;
+    hipThreadIdx_y_channel.z = hipThreadIdx_y + 32;
+
+    uchar *src_lds_channel[3];
```

Comment:
can you switch to uchar3 vector-based here?

Reply:
I actually need an array of 3 pointers -> 3 (uchar*)s. But (uchar3*) gives a single uchar3 pointer.